### DISCUSSION TITLE
Zoom event not called

### DISCUSSION THREAD

**Discussion by Smurfj3** [Mapsui/Mapsui#2154] [Q&A]
I noticed that if I subscribe to the MapView.Zoomed handler it isn't called if you zoom in, at least not when you zoom in using the zoom widget buttons. Is this intended? I tried to find a way around and noticed the ViewPort changed event but that gets called on panning too, so I could use that but it seems rather ineffecient.

**Comment by pauldendulk** [comment]
@Smurfj3  There is probably a bug here but I wonder how this should work. Zooming can be done in many ways. In the most general interpretation a zoom could be any change in zoom level. Perhaps in this case this could be handled as just a click event of the button. Would that be enough in your use case? And what is your use case?

**Reply by Smurfj3** [reply]
I'll be honest I can't even remember what I needed the zoom event for, most likely I was just experimenting. I was just wondering if it was intended or if it was a bug. If it is intented, then it is rather confusing, you'd expect the "Zoom Event" to be called whenever you zoom in, regardless of the method you use. I expected there to be a zoom method that handles the zooming and the event, and this method would be called by widget button/scrollwheel/pinching events.

**Reply by pauldendulk** [reply]
From that description I would think events on the buttons would be the best solution.
